### PR TITLE
Allow special macros {uuid} and {database} in a Replicated database ZooKeeper path

### PR DESCRIPTION
**Pull request by vitlibar**
### Changelog category (leave one):
- Improvement

### Changelog entry:
Allow special macros {uuid} and {database} in a Replicated database ZooKeeper path

### CODE REVIEW

**Comment by robot-clickhouse-ci-1** [comment]
*This is an automated comment for commit e42967c72410d02ed116f7277ec1bcfb49be20f8 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/62818/e42967c72410d02ed116f7277ec1bcfb49be20f8/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

File: src/Databases/DatabaseReplicated.cpp [hunk 1]
```diff
@@ -1708,9 +1708,18 @@ void registerDatabaseReplicated(DatabaseFactory & factory)
         String shard_name = safeGetLiteralValue<String>(arguments[1], "Replicated");
         String replica_name  = safeGetLiteralValue<String>(arguments[2], "Replicated");
 
-        zookeeper_path = args.context->getMacros()->expand(zookeeper_path);
-        shard_name = args.context->getMacros()->expand(shard_name);
-        replica_name = args.context->getMacros()->expand(replica_name);
+        /// Expand macros.
+        Macros::MacroExpansionInfo info;
+        info.table_id.database_name = args.database_name;
+        info.table_id.uuid = args.uuid;
```
Comment (alexey-milovidov) [inline src/Databases/DatabaseReplicated.cpp:1714]: It could be confusing, because here it is the database UUID, not table UUID. But passable.

**Review by alexey-milovidov — APPROVED** [review]

**Comment by alexey-milovidov** [comment]
I don't understand why performance tests showed something. Need to check.

**Comment by vitlibar** [comment]
> I don't understand why performance tests showed something. Need to check.

I only changed `registerDatabaseReplicated()` in this PR, there must not be any impact on performance.

**Pull request merged**